### ISSUE TITLE
awscdk/no-variable-construct-id should only apply when a construct is created in a class constructor

### ISSUE THREAD

**Issue by JohnnyOu** [ren-yamanashi/eslint-cdk-plugin#357]
### Feature Description

According to the [documentation](https://eslint-plugin-awscdk.dev/rules/no-variable-construct-id):
> This rule does not apply to loop processing such as for, while, forEach, map, etc.

However, currently if the construct is created in a method, and the method is called within a loop or in any other way, then this rule flags the usage of variables in ID as a violation.

The intention of abstracting resource creation in a method is to avoid code duplication and to reuse the code multiple times within a stack, similar to the intention of using a for-loop. This kind of usage necessitates distinct construct IDs. For this reason, I believe that the rule `awscdk/no-variable-construct-id` should not apply in class methods that are not the constructor.

### Invalid Code Examples

No new invalid cases need to be introduced.

### Valid Code Examples

```typescript
class MyStack {
  constructor(parent: App, id: string, props: MyStackProps) {
    this.createMyResource("id1");
    this.createMyResource("id2");
    ["id3", "id4", "id5"].forEach((id) => createMyResource(id));
  }

  createMyResource(id: string) {
    new MyConstruct(this, id, { ...someProps });
  }
}
```

### Rule Type

suggestion (improve code)

### Implementation Details

- [x] This rule is related to AWS CDK best practices
- [x] I'm willing to submit a PR for this feature
